src/index.ts

```diff
@@ -53,22 +54,24 @@ export class GraphQLServer {
     if (props.schema) {
       this.executableSchema = props.schema
     } else if (props.typeDefs && props.resolvers) {
-      let { directiveResolvers, typeDefs, resolvers } = props
+      let uploadMixin = {}
+      let { directiveResolvers, resolvers, typeDefs } = props
+
+      if (typeof typeDefs === 'string') {
+        // read from .graphql file if path provided
+        if (typeDefs.endsWith('graphql')) {
+          const schemaPath = path.resolve(typeDefs)
 
-      // read from .graphql file if path provided
-      if (typeDefs.endsWith('graphql')) {
-        const schemaPath = path.resolve(typeDefs)
+          if (!fs.existsSync(schemaPath)) {
+            throw new Error(`No schema found for path: ${schemaPath}`)
+          }
 
-        if (!fs.existsSync(schemaPath)) {
-          throw new Error(`No schema found for path: ${schemaPath}`)
+          typeDefs = importSchema(schemaPath)
         }
 
-        typeDefs = importSchema(schemaPath)
+        uploadMixin = typeDefs.includes('scalar Upload') && { Upload: GraphQLUpload }
```

Comment:
I worry this way of implementing this feature is problematic since the upload types are now just injected in the case where `typeDefs` is a string.

I'd strongly recommend to "merge" the `typeDefs` array into a string (probably by re-implementing a bunch of things done in `graphql-tools`) so you don't need to touch any other code.

Reply:
Can you take a look at https://github.com/graphcool/graphql-yoga/pull/187/commits/c77934edbc4f0f134eac2caac602f74815692e72 ?

This implementation we first map typeDefs array into a single string then run code exactly as it is.

Reply:
Yeah, I like that better!